modules/apps/content-dashboard/content-dashboard-test/src/testIntegration/java/com/liferay/content/dashboard/web/internal/portlet/test/util/MockLiferayPortletRenderRequestUtil.java

```diff
@@ -0,0 +1,111 @@
+import com.liferay.portal.kernel.security.permission.PermissionThreadLocal;
+import com.liferay.portal.kernel.test.portlet.MockLiferayPortletRenderRequest;
+import com.liferay.portal.kernel.test.portlet.MockLiferayPortletURL;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.ProxyUtil;
+import com.liferay.portal.kernel.util.StringBundler;
+import com.liferay.portal.kernel.util.WebKeys;
+
+import java.util.Locale;
+import java.util.Objects;
+
+import javax.portlet.PortletContext;
+import javax.portlet.PortletPreferences;
+import javax.portlet.PortletRequestDispatcher;
+
+/**
+ * @author David Arques
+ */
+public class MockLiferayPortletRenderRequestUtil {
+
+	public static MockLiferayPortletRenderRequest
+			getMockLiferayPortletRenderRequest(
+				String[] assetVocabularyNames, Company company, Locale locale)
+		throws Exception {
+
+		MockLiferayPortletRenderRequest mockLiferayPortletRenderRequest =
+			new MockLiferayPortletRenderRequest();
+
+		mockLiferayPortletRenderRequest.setAttribute(
+			WebKeys.COMPANY_ID, company.getCompanyId());
+
+		mockLiferayPortletRenderRequest.setAttribute(
+			StringBundler.concat(
+				mockLiferayPortletRenderRequest.getPortletName(), "-",
+				WebKeys.CURRENT_PORTLET_URL),
+			new MockLiferayPortletURL());
+
+		String path = "/view.jsp";
+
+		mockLiferayPortletRenderRequest.setParameter("mvcPath", path);
+
+		mockLiferayPortletRenderRequest.setAttribute(
+			MVCRenderConstants.
+				PORTLET_CONTEXT_OVERRIDE_REQUEST_ATTIBUTE_NAME_PREFIX + path,
+			ProxyUtil.newProxyInstance(
+				PortletContext.class.getClassLoader(),
+				new Class<?>[] {PortletContext.class},
+				(PortletContextProxy, portletContextMethod,
+				 portletContextArgs) -> {
+
+					if (Objects.equals(
+							portletContextMethod.getName(),
+							"getRequestDispatcher") &&
+						Objects.equals(portletContextArgs[0], path)) {
+
+						return ProxyUtil.newProxyInstance(
+							PortletRequestDispatcher.class.getClassLoader(),
+							new Class<?>[] {PortletRequestDispatcher.class},
+							(portletRequestDispatcherProxy,
+							 portletRequestDispatcherMethod,
+							 portletRequestDispatcherArgs) -> null);
+					}
+
+					throw new UnsupportedOperationException();
+				}));
+
+		mockLiferayPortletRenderRequest.setAttribute(
+			WebKeys.THEME_DISPLAY, _getThemeDisplay(company, locale));
+
+		PortletPreferences portletPreferences =
+			mockLiferayPortletRenderRequest.getPreferences();
+
+		portletPreferences.setValues(
+			"assetVocabularyNames", assetVocabularyNames);
+
+		return mockLiferayPortletRenderRequest;
+	}
+
+	private static ThemeDisplay _getThemeDisplay(Company company, Locale locale)
+		throws Exception {
+
+		ThemeDisplay themeDisplay = new ThemeDisplay();
+
+		themeDisplay.setCompany(company);
+		themeDisplay.setLocale(locale);
+		themeDisplay.setPermissionChecker(
+			PermissionThreadLocal.getPermissionChecker());
```

Comment:
You should be very carefulll with using in a Util test class PermissionThreadLocal.getPermissionChecker(), because the PermissionThreadLocal should be initialized, and you can't be sure what is the permission checker that you are using here.

Reply:
AS I said before, this class just created to be used in these 2 tests. And both have the same behavior in this regard

Reply:
This is not a problem anymore